### PR TITLE
Add page for creating new and editing existing baseline

### PR DESCRIPTION
**Pull request by johnsonm325**

### CODE REVIEW

File: src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js [hunk 1]
```diff
@@ -1,128 +1,410 @@
...
+                ),
+            renderEdit: (value, { column, columnIndex, rowIndex, rowData }, { activeEditId }) => {
+                const renderer = rowData.hasOwnProperty('parent') ? childEditRenderer : parentEditRenderer;
+                return renderer(value, { rowData, column, rowIndex, columnIndex, activeEditId });
+            }
+        });
 
-        if (!baselineData) {
-            rows = [];
+        /*eslint-disable react/jsx-key*/
```
Comment (beav) [inline src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js]: can this be rewritten to remove the disable?

File: src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js [hunk 2]
```diff
@@ -1,128 +1,410 @@
...
-        let rows = [];
-        let rowData = [];
+        this.changeBaselineName = value => {
+            this.setState({ value });
+        };
 
-        for (let i = 0; i < 2; i += 1) {
-            rowData.push(<Skeleton size={ SkeletonSize.md } />);
-        }
+        this.submitBaselineName = (/*baselineName*/) => {
```
Comment (beav) [inline src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js]: dunno if this inline comment is intentional or not

File: src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js [hunk 3]
```diff
@@ -1,128 +1,410 @@
...
+                return { rows, activeEditId: null };
+            });
+        };
+
+        this.buildApiBody = (rows, rowIndex) => {
+            let row = rows[rowIndex];
+            let apiBody = {};
+            let baselineFactsArray = [];
+
+            /*eslint-disable camelcase*/
```
Comment (beav) [inline src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js]: can the camelcase be rewritten with `_`'s? parent_row for example

**Comment by beav** [comment]
overall looks ok, will test in a bit

File: src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js (hunk 2 above)
Comment (johnsonm325) [reply src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js]: I'm leaving it there for when I do the next phase of this.

File: src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js (hunk 3 above)
Comment (johnsonm325) [reply src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js]: This disables the requirement of camelcase. It's because of baseline_facts.

**Comment by beav** [comment]
this has a couple of issues but I'd rather treat them as bugs than let this PR get bigger.

ACK

**Pull request merged**